packages/odyssey-react-mui/src/Tag.tsx

```diff
@@ -31,8 +48,142 @@ export type TagProps = {
+      deleteIcon: odysseyDesignTokens.HueAccentTwo500,
+      deleteIconHover: odysseyDesignTokens.HueAccentTwo600,
+    },
+    accentThree: {
+      background: odysseyDesignTokens.HueAccentThree100,
+      hover: odysseyDesignTokens.HueAccentThree200,
+      active: odysseyDesignTokens.HueAccentThree300,
+      text: odysseyDesignTokens.HueAccentThree700,
+      border: odysseyDesignTokens.HueAccentThree200,
+      deleteIcon: odysseyDesignTokens.HueAccentThree500,
+      deleteIconHover: odysseyDesignTokens.HueAccentThree600,
+    },
+    accentFour: {
+      background: odysseyDesignTokens.HueAccentFour100,
+      hover: odysseyDesignTokens.HueAccentFour200,
+      active: odysseyDesignTokens.HueAccentFour300,
+      text: odysseyDesignTokens.HueAccentFour700,
+      border: odysseyDesignTokens.HueAccentFour200,
+      deleteIcon: odysseyDesignTokens.HueAccentFour500,
+      deleteIconHover: odysseyDesignTokens.HueAccentFour600,
+    },
+  };
+
+  return colors[colorVariant] || colors.default;
+};
+
+const StyledTag = styled(MuiChip, {
+  shouldForwardProp: (prop) =>
+    !["colorVariant", "odysseyDesignTokens"].includes(prop as string),
+})<{
+  colorVariant: TagColorVariant;
+  odysseyDesignTokens: DesignTokens;
+  as?: React.ElementType; // Allow the 'as' prop to be forwarded
+}>(({ colorVariant, odysseyDesignTokens }) => {
+  const colors = getChipColors(colorVariant, odysseyDesignTokens);
+
+  return {
+    height: "auto",
+    paddingBlock: `calc(${odysseyDesignTokens.Spacing2} - ${odysseyDesignTokens.BorderWidthMain})`,
+    paddingInline: odysseyDesignTokens.Spacing3,
+    fontSize: odysseyDesignTokens.TypographySizeBody,
+    lineHeight: odysseyDesignTokens.TypographyLineHeightUi,
+    borderRadius: odysseyDesignTokens.BorderRadiusRound,
+    borderWidth: odysseyDesignTokens.BorderWidthMain,
+    borderStyle: odysseyDesignTokens.BorderStyleMain,
+    backgroundColor: colors.background,
+    color: colors.text,
+    borderColor: colors.border,
+
+    "&.MuiChip-clickable:hover": {
+      backgroundColor: colors.hover,
+    },
+
+    "&.MuiChip-clickable:active": {
+      backgroundColor: colors.active,
+    },
+
+    "&.Mui-disabled": {
+      opacity: 1,
+      pointerEvents: "none",
+      borderColor: odysseyDesignTokens.BorderColorDisabled,
+      color: odysseyDesignTokens.TypographyColorDisabled,
+
+      "& .MuiChip-deleteIcon": {
+        color: odysseyDesignTokens.HueNeutral300,
+      },
+    },
+
+    "& .MuiChip-icon": {
+      margin: 0,
+      marginInlineEnd: odysseyDesignTokens.Spacing1,
+    },
+
+    "& .MuiChip-deleteIcon": {
+      WebkitTapHighlightColor: "transparent",
+      color: colors.deleteIcon,
+      fontSize: "1em",
+      cursor: "pointer",
+      margin: 0,
+      marginInlineStart: odysseyDesignTokens.Spacing2,
+
+      "&:hover": {
+        color: colors.deleteIconHover,
+      },
+    },
+
+    "& .MuiChip-label": {
+      padding: 0,
+    },
+  };
```

Comment:
@benlister-okta I love the idea of pulling styles out of components.tsx but, in this case, I think we keep the 'non-color' related styles in components.tsx. I'm worried that moving them may have some unintended consequences. I do think anything color variant related should be in here though